pkg/ingester/ingester.go

```diff
@@ -714,16 +725,54 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 	)
 
 	// Walk the samples, appending them to the users database
-	app := db.Appender(ctx).(extendedAppender)
-	level.Debug(spanlog).Log("event", "got appender", "numSeries", len(req.Timeseries))
+	var persistentApp, ephemeralApp extendedAppender
 
-	var activeSeries *activeseries.ActiveSeries
-	if i.cfg.ActiveSeriesMetricsEnabled {
-		activeSeries = db.activeSeries
+	rollback := func() {
+		if persistentApp != nil {
+			if err := persistentApp.Rollback(); err != nil {
+				level.Warn(i.logger).Log("msg", "failed to rollback on error", "user", userID, "err", err)
```

Comment:
[nit] I would clarify it.

```suggestion
				level.Warn(i.logger).Log("msg", "failed to rollback persistent appender on error", "user", userID, "err", err)
```